starter-code/scripts/models/article.js

```diff
@@ -89,13 +102,27 @@ Article.numWordsByAuthor = function() {
     return Article.allAuthors().map(function(currentAuthor) {
       return {
         name: currentAuthor,
-        numWords: // someCollection.filter(function(curArticle) {
+        numWords: Article.allArticles.map(function(currentArticle) {
+          if (currentArticle.author === currentAuthor) {
+            return currentArticle.body.match(/\w+/g).length
+          }
+          else {
+            return null
+          }
+        })
```

Comment:
Great work limiting the articles of interest to those written by the current author and obtaining word counts for all of the relevant articles! Alternatively, you could have filtered the allArticles array first to only include articles by the current author, and then mapped through that array, obtaining the word counts of those articles (instead of setting the value to 0 in the array).